conformance/lang/expressions/trap_expr.balt

```diff
@@ -0,0 +1,351 @@
+    ()|error t1 = trap ();
+    io:println(t1 is ()); // @output true
+
+    error? t2 = trap null;
+    io:println(t2 is ()); // @output true
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a boolean-literal and it does not panic.
+Labels: boolean, boolean-literal, error-type, trap-expr, union-type
+
+function init() {
+    boolean|error t = trap false;
+    io:println(t); // @output false
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a numeric-literal and it does not panic.
+Labels: decimal, error-type, float, int, trap-expr, union-type
+
+function init() {
+    int|error t1 = trap 2;
+    io:println(t1); // @output 2
+    float|error t2 = trap 2e3f;
+    io:println(t2); // @output 2000.0
+    decimal|error t3 = trap 2e355d;
+    io:println(t3); // @output 2E+355
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a string-literal and it does not panic.
+Labels: string, trap-expr, union-type
+
+function init() {
+    string|error t1 = trap "Hello";
+    io:println(t1); // @output Hello
+    t1 = trap "Hello\tWorld";
+    io:println(t1); // @output Hello	World
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a byte-array-literal and it does not panic.
+Labels: byte, byte-array-literal, error-type, trap-expr, union-type
+
+function init() {
+    byte[]|error t1 = trap base16 `aeeecdefabcd12345567888822`;
+    io:println(t1); // @output [174,238,205,239,171,205,18,52,85,103,136,136,34]
+    t1 = trap base64 `aGVsbG8gYmFsbGVyaW5hICEhIQ==`;
+    io:println(t1); // @output [104,101,108,108,111,32,98,97,108,108,101,114,105,110,97,32,33,33,33]
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a string-template-expr and it does not panic.
+Labels: error-type, string-template-expr, trap-expr, union-type
+
+function init() {
+    string|error t1 = trap string `Hello World!`;
+    io:println(t1); // @output Hello World!
+    string s = string `Hello World!`;
+    string|error t2 = trap string `${s}`;
+    io:println(t2); // @output Hello World!
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a xml-template-expr and it does not panic.
+Labels: error-type, trap-expr, union-type, xml-template-expr
+
+function init() {
+    xml|error t1 = trap xml `<Greeting>Hello!</Greeting>`;
+    io:println(t1); // @output <Greeting>Hello!</Greeting>
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a list constructor expression. Test 1.
+Labels: array-type, error-type, int, list-constructor-expr, string, trap-expr, union-type
+
+function init() {
+    int[]|error t1 = trap [1, 2, 3, 4];
+    io:println(t1); // @output [1,2,3,4]
+    (int|string)[]|error t2 = trap [1, "2", 3, "4"];
+    io:println(t2); // @output [1,"2",3,"4"]
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a list constructor expression. Test 2.
```

Comment:
Isn't the difference between this and the previous the fact that one doesn't complete normally? Why aren't we saying that instead of Test 1 and Test 2?